src/main/java/org/pulseus/auth/service/rest/controller/CSVController.java

```diff
@@ -0,0 +1,114 @@
+package org.pulseus.auth.service.rest.controller;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.util.Calendar;
+import java.util.Locale;
+import java.util.UUID;
+
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.liferay.portal.kernel.configuration.Configuration;
+import com.liferay.portal.kernel.configuration.ConfigurationFactoryUtil;
+import com.liferay.portal.kernel.model.Organization;
+import com.liferay.portal.kernel.model.Role;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.security.auth.CompanyThreadLocal;
+import com.liferay.portal.kernel.service.OrganizationLocalServiceUtil;
+import com.liferay.portal.kernel.service.RoleLocalServiceUtil;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.UserLocalServiceUtil;
+import com.liferay.portal.kernel.util.PortalClassLoaderUtil;
+import com.liferay.portal.kernel.util.StringPool;
+
+@RestController
+public class CSVController {
+	Configuration configuration = ConfigurationFactoryUtil.getConfiguration(PortalClassLoaderUtil.getClassLoader(), "portlet");
+
+	private static final Logger log = LogManager.getLogger(JWTController.class);
+
+	@RequestMapping(value = "/addusers", method = RequestMethod.GET)
+	public void addusers() throws Exception {
+		System.out.println("adding users");
+		ServiceContext serviceContext = new ServiceContext();
+		Long companyId = CompanyThreadLocal.getCompanyId();	
+
+
+
+		BufferedReader br = null;
+		String line = "";
+		String cvsSplitBy = ",";
+
+		br = new BufferedReader(new FileReader("/opt/pulse/users.csv"));
+		while (true) {
+			try {
+				if ((line = br.readLine()) != null) {
+					// use comma as separator
+					String[] value = line.split(cvsSplitBy);
+					Role role = RoleLocalServiceUtil.getRole(companyId, value[5]);
+					Long roleId = 	role.getRoleId();
+
+					Organization org1 = OrganizationLocalServiceUtil.getOrganization(companyId, value[6]);
+					Long orgId1 = org1.getOrganizationId();
+
+					Organization org2 = OrganizationLocalServiceUtil.getOrganization(companyId, value[7]);
+					Long orgId2 = org2.getOrganizationId();
+
+					long creatorUserId = 0;
```

Comment:
All of these are assignments that should happen outside of the loop; they could be set once as static values and not reallocated each time through